astropy/coordinates/angles.py

```diff
@@ -644,12 +650,20 @@ def _wrap_internal(self):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
+        # TODO: This could probably be sped up as well.  u.degree.to(self.unit)
+        # should be faster if self.unit *is* u.degree
         a360 = u.degree.to(self.unit, 360.0)
-        wrap_angle = self.wrap_angle.to(self.unit).value
+
+        # TODO: Quantity.to is currently slow when the quantity is already in
+        # the requested unit.
+        if self.wrap_angle.unit is self.unit:
+            wrap_angle = self.wrap_angle.value
+        else:
+            wrap_angle = self.wrap_angle.to(self.unit).value
         wrap_angle_floor = wrap_angle - a360
         self_angle = self.value
         # Do the wrapping, but only if any angles need to be wrapped
-        if np.any(self_angle < wrap_angle_floor) or np.any(self_angle >= wrap_angle):
+        if np.any((self_angle < wrap_angle_floor) | (self_angle >= wrap_angle)):
```

Comment:
Interestingly, instantiating an `Angle` from an existing `Angle` is no faster than instantiating one from a `Quantity` (such as `360 * u.deg`).  In fact it may even be a smidge slower.  That's not including the instantiation of the `Quantity` itself.  We're talking `Angle(a)` vs. `Angle(q)`.